core-processor/src/processor.rs

```diff
@@ -226,6 +226,7 @@ fn process_success(
 
     match kind {
         Exit(value_destination) => {
+            journal.push(JournalNote::MessageConsumed(message_id));
```

Comment:
I would refactor this matching like this:
```rust
if let Wait = kind {
    journal.push(JournalNote::WaitDispatch(
        dispatch.into_stored(program_id, context_store),
    ));
} else {
    if let Exit(value_destination) = kind {
        journal.push(JournalNote::ExitDispatch {
            id_exited: program_id,
            value_destination,
        });
    }

    let outcome = match dispatch.kind() {
        DispatchKind::Init => DispatchOutcome::InitSuccess {
            message_id,
            origin,
            program_id,
        },
        _ => DispatchOutcome::Success(message_id),
    };

    journal.push(JournalNote::MessageDispatched(outcome));
    journal.push(JournalNote::MessageConsumed(message_id));
}
```

Cause MessageDispatched should also be deposited.
Or split it by adding one more DispatchOutcome to show that the message was exited, so program is no more available.

Reply:
Yeah, the second variant with propagating new event seems better!

Reply:
> Yeah, the second variant with propagating new event seems better!

make sense, btw, should we `remove_waiting_message` in `exit_dispatch` also? 

https://github.com/gear-tech/gear/blob/3981f45830a519b68fce27be8767823f2e3497d5/pallets/gear/src/manager.rs#L318

exit_dispatch now only

- set_program_terminated_status
- remove_program_waitlist